web_ui_awareness.md

```diff
@@ -1,42 +1,146 @@
+
+- We can also learn from past challenges in getting teams to adopt Tilt. We've been successful engaging implementers who get Tilt to work, but less so in selling their teammates. Can we help "show, not tell" the value of Tilt with our Web UI?
 
+
+
+&nbsp;
 ## Requirements
 
-- You can see that Tilt is reacting to file system events.
-- When you save a file, Tilt should communicate that it saw the save.
-- You san see that Tilt is reacting to cluster events.
-- You should be able to find the pod ID in the cluster.
-- Tilt should communicate when a service has an endpoint URL that you can open.
-- You should be able to see what Tilt is doing Right Now, with some explanation/justification for why it's doing that.
-- Errors should pop into your view. You shouldn't have to remember to look in an error bucket.
+Changes to the Sidebar, Statusbar, and a new Error pane make it more visible when Tilt responds to file system updates, builds/updates and deploys your resources, and encounters exceptions. 
 
-## Non-Requirements
+Since this is the first pass at a standalone Web UI, we limit our focus to two common use cases: 
+
+1. Normal iterative flow. You can see Tilt is working properly.
+
+2. Errors. You see when and where the error occurred, and see log context.
 
-These are things that are non-essential right now, but which we may want to tackle in the future.
 
-- The build history of an image
-- The duration of an image build
-- The YAML Tilt read
-- The YAML Tilt deployed
+&nbsp;
+## Non-Requirements
+Tilt has lots of potentially relevant and useful data that we don't show (or surface) in the Web UI. We need more feedback on this first pass, but we should eventually think through whether or how to add other data to our UI:
+
+- Build history for individual resources
+- The YAML Tilt read or deployed
 - The history of changes to a Pod (e.g., `kubectl get pods -w`)
 
-## Detailed design
+We can also better explain specific types of errors, rather than let the user infer:
+
+- All images depend on one base image, and there’s an error there.
+- Your cluster is down. Or your Internet is down.
+
+
+&nbsp;
+## Detailed Design
+
+In the default interface, the **overview** panes allow you to navigate to **detail** panes, which either show more information about a single resource (Ex: `Sidebar > doggos > Log`), or show a custom filtered view of your data (Ex: `Sidebar > ALL > Errors`).
+
+***
+The changes support these use cases, which assume you already have a Tiltfile set up:
+
+1. You change code, then see a successful rebuild and deploy
+    - You see Tilt detect the change and start working (and what triggered it)
+    - You see Tilt is done (and know your service is ready to check)
+
+2. You introduce errors, see what happened, and how to resolve them.
+    - Build error
+    - Runtime error (after you trigger the error)
```

Comment:
you're missing "startup error" (sort of a subcategory of runtime error but we've been discussing them separately)